numba/targets/npyimpl.py

```diff
@@ -540,3 +541,47 @@ def generate(self, *args):
 
 
 del _kernels
+
+@intrinsic
+def _make_dtype_object(typingctx, desc):
+    """Given a string description *desc*, returns the dtype object.
+    """
+    if isinstance(desc, types.Const):
+        # Convert the str description into np.dtype then to numba type.
+        nb_type = from_dtype(np.dtype(desc.value))
+        return_type = types.DType(nb_type)
+        sig = return_type(desc)
+
+        def codegen(context, builder, signature, args):
```

Comment:
Can these identical impls be lifted out of the condition?

Reply:
Done.